doc/inband-telemetry/SAI-Proposal-Advanced-Telemetry-draft.md

```diff
@@ -0,0 +1,421 @@
+        _In_ sai_object_id_t *telemetry_report_trigger_id);
+
+typedef sai_status_t (*sai_telemetry_report_trigger_attr_get_fn)(
+        _In_    sai_object_id_t telemetry_report_trigger_id,
+        _In_    uint32_t attr_count,
+        _Inout_ const sai_attribute_t *attr_list);
+
+typedef sai_status_t (*sai_telemetry_report_trigger_attr_set_fn)(
+        _In_  sai_object_id_t telemetry_report_trigger_id,
+        _In_  uint32_t attr_count,
+        _In_  const sai_attribute_t *attr_list);
+
+typedef sai_status_t (*sai_telemetry_int_session_create_fn)(
+        _Out_ sai_object_id_t *telemetry_int_session_id,
+        _In_  uint32_t attr_count,
+        _In_  const sai_attribute_t *attr_list);
+
+typedef sai_status_t (*sai_telemetry_int_session_remove_fn)(
+        _In_ sai_object_id_t *telemetry_int_session_id);
+
+typedef sai_status_t (*sai_telemetry_int_session_attr_get_fn)(
+        _In_    sai_object_id_t telemetry_int_session_id,
+        _In_    uint32_t attr_count,
+        _Inout_ const sai_attribute_t *attr_list);
+
+typedef sai_status_t (*sai_telemetry_int_session_attr_set_fn)(
+        _In_  sai_object_id_t telemetry_int_session_id,
+        _In_  uint32_t attr_count,
+        _In_  const sai_attribute_t *attr_list);
+
+typedef sai_status_t (*sai_telemetry_int_sink_port_create_fn)(
+        _Out_ sai_object_id_t *telemetry_int_sink_port_id,
+        _In_  uint32_t attr_count,
+        _In_  const sai_attribute_t *attr_list);
+
+typedef sai_status_t (*sai_telemetry_int_sink_port_remove_fn)(
+        _In_ sai_object_id_t *telemetry_int_sink_port_id);
+
+typedef struct _sai_telemetry_api_t {
+    sai_telemetry_create_fn     telemetry_create;
+    sai_telemetry_remove_fn     telemetry_remove;
+    sai_telemetry_attr_get_fn   telemetry_attr_get;
+    sai_telemetry_attr_set_fn   telemetry_attr_set;
+
+    sai_telemetry_watchlist_entry_create_fn   telemetry_watchlist_entry_create;
+    sai_telemetry_watchlist_entry_remove_fn   telemetry_watchlist_entry_remove;
+    sai_telemetry_watchlist_entry_attr_get_fn telemetry_watchlist_entry_attr_get;
+    sai_telemetry_watchlist_entry_attr_set_fn telemetry_watchlist_entry_attr_set;
+
+    sai_telemetry_report_trigger_create_fn    telemetry_report_trigger_create;
+    sai_telemetry_report_trigger_remove_fn    telemetry_report_trigger_remove;
+    sai_telemetry_report_trigger_attr_get_fn  telemetry_report_trigger_attr_get;
+    sai_telemetry_report_trigger_attr_set_fn  telemetry_report_trigger_attr_set;
+
+    sai_telemetry_int_session_create_fn     telemetry_int_session_create;
+    sai_telemetry_int_session_remove_fn     telemetry_int_session_remove;
+    sai_telemetry_int_session_attr_get_fn   telemetry_int_session_attr_get;
+    sai_telemetry_int_session_attr_set_fn   telemetry_int_session_attr_set;
+
+    sai_telemetry_int_sink_port_create_fn   telemetry_int_sink_port_create;
+    sai_telemetry_int_sink_port_remove_fn   telemetry_int_sink_port_remove;
```

Comment:
missing set/get functions for telemetry_int_sink_port (we need quad api even if those functions will be dummy)